inmoose/edgepy/exactTestBetaApprox.py

```diff
@@ -0,0 +1,90 @@
+# -----------------------------------------------------------------------------
+# Copyright (C) 2008-2022 Yunshun Chen, Aaron TL Lun, Davis J McCarthy, Matthew E Ritchie, Belinda Phipson, Yifang Hu, Xiaobei Zhou, Mark D Robinson, Gordon K Smyth
+# Copyright (C) 2024 Maximilien Colange
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+# This file is based on the file 'R/exactTestBetaApprox.R' of the Bioconductor edgeR package (version 3.38.4).
+
+import numpy as np
+from scipy.stats import beta
+
+
+def exactTestBetaApprox(y1, y2, dispersion=0):
+    """
+    Compute genewise *p*-values for differences in the means between two groups of negative-binomially distributed counts.
+
+    This function implements an asymptotic beta distribution approximation to
+    the conditional count distribution.
+
+    See also
+    --------
+    exactTest
+
+    Arguments
+    ---------
+    y1 : matrix
+        matrix of counts for the first of the two experimental groups to be
+        tested for differences. Rows correspond to genes and columns to
+        libraries. Libraries are assumed to be equal in size -- *e.g.* adjusted
+        pseudocounts from the output of :func:`equalizeLibSizes`.
+    y2 : matrix
+        matrix of counts for the second of the two experimental groups to be
+        tested for differences. Rows correspond to genes and columns to
+        libraries. Libraries are assumed to be equal in size -- *e.g.* adjusted
+        pseudocounts from the output of :func:`equalizeLibSizes`.
+    dispersion : array_like of floats
```

Comment:
Here you have array_like of floats as type but the default value is int. Is it normal?

Reply:
Not a big deal, as the `int`s will be converted when combined with the other `float` values.
Fixed the default value to `0.` though.